### PR TITLE
feat(webpack): Upgrade mini-css-extract-webpack-plugin and support hmr

### PR DESCRIPTION
**Pull request by liril-net**
#565

### CODE REVIEW

File: core/poi/lib/plugins/config-css.js [hunk 1]
```diff
@@ -64,6 +64,10 @@ exports.apply = api => {
             rule
               .use('extract-css-loader')
               .loader(require('mini-css-extract-plugin').loader)
+              .options({
+                hmr: !api.isProd,
```
Comment (egoist) [inline core/poi/lib/plugins/config-css.js]:
```suggestion
                hmr: api.mode === 'development',
```

Only need it in `development` mode, FYI there's also `test` mode

**Comment by egoist** [comment]
We use Yarn 😄

File: core/poi/package.json [hunk 2]
```diff
@@ -54,7 +54,7 @@
     "joycon": "^2.2.1",
     "launch-editor-middleware": "^2.2.1",
     "lodash.merge": "^4.6.1",
-    "mini-css-extract-plugin": "^0.4.5",
+    "mini-css-extract-plugin": "^0.7.0",
```
Comment (egoist) [inline core/poi/package.json]: yarn.lock needs to be updated too

File: core/poi/lib/plugins/config-css.js [hunk 3]
```diff
@@ -64,6 +64,10 @@ exports.apply = api => {
             rule
               .use('extract-css-loader')
               .loader(require('mini-css-extract-plugin').loader)
+              .options({
+                hmr: api.mode === 'development',
+                reloadAll: true
+              })
```
Comment (egoist) [inline core/poi/lib/plugins/config-css.js]: I just realized that we need to enable css extraction for non-production mode as well.

Comment (liril-net) [reply core/poi/lib/plugins/config-css.js]: Oops, so this feature is not practical.

Comment (egoist) [reply core/poi/lib/plugins/config-css.js]: just make `css.extract` default to `true` in development mode as well.

Comment (liril-net) [reply core/poi/lib/plugins/config-css.js]: But, is css extract useful in development mode?

File: core/poi/lib/plugins/config-css.js [hunk 4]
```diff
@@ -6,7 +6,9 @@ exports.cli = ({ command, isProd }) => {
   if (isProd) {
     command.option('--no-extract-css', `Don't extract CSS files`)
   } else {
-    command.option('--extract-css', 'Extract CSS files')
+    command.option('--extract-css', 'Extract CSS files', {
+      default: true
```
Comment (egoist) [inline core/poi/lib/plugins/config-css.js]: for some reason we can't set default value for cli flags, instead we do it here: https://github.com/egoist/poi/blob/e4b5b04a3f0c4c278fd4b277e609bb52423056bf/core/poi/lib/utils/validateConfig.js#L77

**Comment by egoist** [comment]
seems hmr doesn't work for css modules.

File: core/poi/lib/plugins/config-css.js [hunk 5]
```diff
@@ -64,6 +64,10 @@ exports.apply = api => {
             rule
               .use('extract-css-loader')
               .loader(require('mini-css-extract-plugin').loader)
+              .options({
+                hmr: api.mode === 'development',
+                reloadAll: true
```
Comment (MarchWorks) [inline core/poi/lib/plugins/config-css.js]: is `reloadAll` necessary?

**Pull request merged**